metaflow/plugins/aws/step_functions/step_functions_decorator.py

```diff
@@ -20,7 +21,11 @@ def task_pre_step(self,
                       graph,
                       retry_count,
                       max_user_code_retries,
-                      ubf_context):
+                      ubf_context,
+                      input_obj):
+        if ubf_context == UBF_TASK:
+            # Don't register any metadata for tasks that don't run within SFN.
```

Comment:
Can you elaborate a bit more?  I'm not sure I (or future generations reading this code) would understand the link between UBF and SFN and metadata here

Reply:
Adding to @oavdeev 's comment - I don't think we need this check here.

Reply:
This is in the Meson implementation. Happy to strip it. I don't know why it got skipped when we ported to SFN. I figured that since the only UBF is in Meson, it's probably the correct one but I don't care either way.